### DISCUSSION TITLE
Is there a way to "snap" windows in Rectangle Pro or any other Ryan Hanson utility?

### DISCUSSION THREAD

**Discussion by lpridpath** [rxhanson/RectanglePro-Community#189] [Q&A] [answered]
**Firstly** 

I love Rectangle Pro. Thank you SO much for creating such a useful tool.

**Secondly**
Apologies if this is an obvious feature. I am still learning and not fully aware of every functionality so any guidance would be greatly appreciated. 

**Context:**
On my desktop pc, I run [PowerToys](https://learn.microsoft.com/en-us/windows/powertoys/) ([github link](https://github.com/microsoft/PowerToys)) which includes [FanzyZones,](https://learn.microsoft.com/en-us/windows/powertoys/fancyzones) a popular Microsoft Windows window manager utility for arranging and snapping windows.

**Question: (and this might turn into a proposed feature request depending on the answer)**
Pretty much the title... but is there a way to "snap" windows / applications to the zones that are presented in the "Snap Areas" section? I know I can drag them to zones or "areas" as they are referred to, but only the present zone regarding where my mouse was dragged to pops up.

**Use Case / Value:**
I feel like this functionality is very close to what is already implemented in Rectangle, so I might be journeying down the path to posting a feature request. I did see that currently feature requests are paused at this time, and understand the simplistic vision that is meant for Rectangle, but I could see this being a great addition and one that adds this functionality to the lacking native MacOS, and brings parity to other OS's. And all this being available from Rectangle Pro.

**Thanks!**
Any answers, discussion, or input would be greatly appreciated. 

Best.

**Comment by rxhanson** [comment, marked as answer]
Thanks for the feedback! 

Right now in Rectangle Pro, you can build out functionality that is similar to Fancy Zones, although it is a little tedious to configure. To do this, you need to create custom shortcuts for each snap area that you want, and check the box for each one to use it as a "Snap Target". In the "Snap Areas" tab, you can set modifier keys for enabling Snap Targets. I intend to build out a UI specifically for making configuration easier, but just haven't gotten to it yet.

Let me know if you get a chance to try it out and if you get stuck anywhere along the way.

**Reply by lpridpath** [reply]
Thank you so much for the thoughtful and quick response! This's great that this has sort of already been thought about.

It was fun to play around with it based on your direction in your response. I do think it's a little cumbersome to configure and would benefit immensely from UI to create snap areas and configure for optimal use. I think if this were to be fledged out it would provide a new workflow, one that integrates nicely with pre existing Rectangle features. 

Thanks again for your reply. I'm really interested to see this grow but also know you must be swamped with priorities. If it's any encouragement from one random internet stranger, keep chugging along - we appreciate it! I might throw some mock-ups together if it's any interest to you.. or even just for fun myself.

Best.

**Reply by dmberezovskyi** [reply]
@rxhanson, I think a good addition to the Snap Areas feature will be configuring "Only show Snap Targets while holding" with the right mouse click when dragging the window. Like it implemented in the Fancy Zones. It's very comfortable to snap windows without touching the keyboard.

**Reply by rxhanson** [reply]
@dmberezovskyi This makes sense to me; I'll see if I can add it in when next time I'm working on Snap Targets.

**Reply by lpridpath** [reply]
I like the callout and would agree with @dmberezovskyi.

Thanks, @rxhanson! Would be nice if the modifier key could be configurable. I.E. Dmytro suggests the right mouse click, I prefer whilst holding shift.

**Reply by dmberezovskyi** [reply]
@rxhanson, thanks! Will wait for the next updates.
